providers/dns/designate/designate.go

```diff
@@ -62,15 +64,32 @@ type DNSProvider struct {
 // NewDNSProvider returns a DNSProvider instance configured for Designate.
 // Credentials must be passed in the environment variables:
 // OS_AUTH_URL, OS_USERNAME, OS_PASSWORD, OS_TENANT_NAME, OS_REGION_NAME.
+// Or you can specify OS_CLOUD to read the credentials from the according cloud entry.
 func NewDNSProvider() (*DNSProvider, error) {
-	_, err := env.Get(EnvAuthURL, EnvUsername, EnvPassword, EnvTenantName, EnvRegionName)
-	if err != nil {
-		return nil, fmt.Errorf("designate: %w", err)
-	}
+	var opts gophercloud.AuthOptions
+	val, err := env.Get(EnvCloud)
 
-	opts, err := openstack.AuthOptionsFromEnv()
-	if err != nil {
-		return nil, fmt.Errorf("designate: %w", err)
+	if err == nil {
+		cloudOpts, err := clientconfig.AuthOptions(&clientconfig.ClientOpts{
+			Cloud: val[EnvCloud],
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("designate: %w", err)
+		}
+
+		opts = *cloudOpts
+	} else {
+		// TODO this is broken -> variables in _FILE will be ignored!
+		_, err = env.Get(EnvAuthURL, EnvUsername, EnvPassword, EnvTenantName, EnvRegionName)
```

Comment:
It seems like this never worked as intended since we don't use the fetched values and we don't set the variables found in `_FILE` as env. variable. I would propose to remove the feature and instead refer to the `clouds.yaml`. This has multiple benefits: we can drop the hardcoded variables in https://github.com/go-acme/lego/pull/1219/files#diff-91861d95ae59b1d7adb03d128b1c20efL35 and just use the `openstack.AuthOptionsFromEnv()` function to fetch the credentials (this function already does some sanity checks) and we would also get application credentials "for free"